diff -ruN qpy-pp/QtGui/qpyopengl_value_array.cpp qpy/QtGui/qpyopengl_value_array.cpp
--- qpy-pp/QtGui/qpyopengl_value_array.cpp	2020-11-15 23:10:29.343788327 +0100
+++ qpy/QtGui/qpyopengl_value_array.cpp	2020-11-15 23:11:27.997664757 +0100
@@ -208,11 +208,6 @@
             array_type = GL_FLOAT;
             break;
 
-#if defined(SIP_FEATURE_PyQt_Desktop_OpenGL)
-        case 'd':
-            array_type = GL_DOUBLE;
-            break;
-#endif
 
         default:
             PyErr_Format(PyExc_TypeError, "unsupported buffer type '%s'",
@@ -318,14 +313,6 @@
         element_size = sizeof (GLfloat);
         break;
 
-#if defined(SIP_FEATURE_PyQt_Desktop_OpenGL)
-#if GL_DOUBLE != GL_FLOAT
-    case GL_DOUBLE:
-        convertor = convert_double;
-        element_size = sizeof (GLdouble);
-        break;
-#endif
-#endif
 
     default:
         Py_DECREF(seq);
